src/protocol/kademlia/impl/get_value_executor.cpp

```diff
@@ -0,0 +1,328 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <libp2p/protocol/kademlia/impl/get_value_executor.hpp>
+
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/ordered_index.hpp>
+#include <boost/multi_index_container.hpp>
+
+#include <libp2p/protocol/kademlia/config.hpp>
+#include <libp2p/protocol/kademlia/error.hpp>
+#include <libp2p/protocol/kademlia/impl/put_value_executor.hpp>
+#include <libp2p/protocol/kademlia/impl/session.hpp>
+#include <libp2p/protocol/kademlia/message.hpp>
+
+namespace libp2p::protocol::kademlia {
+
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+  std::atomic_size_t GetValueExecutor::instance_number = 0;
+
+  GetValueExecutor::GetValueExecutor(
+      const Config &config, std::shared_ptr<Host> host,
+      std::shared_ptr<Scheduler> scheduler,
+      std::shared_ptr<SessionHost> session_host,
+      std::shared_ptr<PeerRouting> peer_routing,
+      std::shared_ptr<ContentRoutingTable> content_routing_table,
+      const std::shared_ptr<PeerRoutingTable> &peer_routing_table,
+      std::shared_ptr<ExecutorsFactory> executor_factory,
+      std::shared_ptr<Validator> validator, ContentId key,
+      FoundValueHandler handler)
+      : config_(config),
+        host_(std::move(host)),
+        scheduler_(std::move(scheduler)),
+        session_host_(std::move(session_host)),
+        peer_routing_(std::move(peer_routing)),
+        content_routing_table_(std::move(content_routing_table)),
+        executor_factory_(std::move(executor_factory)),
+        validator_(std::move(validator)),
+        key_(std::move(key)),
+        handler_(std::move(handler)),
+        target_(key_),
+        log_("kad", "GetValueExecutor", ++instance_number) {
+    BOOST_ASSERT(host_ != nullptr);
```

Comment:
Not an issue, but why not just ```BOOST_ASSERT(host_)``` and so on? :)

Reply:
I guess it is more human readable =)